pishutdown.py

```diff
@@ -23,14 +27,17 @@ def buttonStateChanged(pin):
 
     if not (GPIO.input(pin)):
         # button is down
-        buttonPressedTime = datetime.now()
+        if buttonPressedTime is None:
```

Comment:
What precisely happened with my cheap button, was that when I released, for some reasons, it both fired a press button event and a release button event. Resulting into systematic reboot, and never shutdown.